src/pipecat/services/speechmatics/stt.py

```diff
@@ -415,60 +313,221 @@ def __init__(
         await super().start(frame)
         await self._connect()
+        self._stt_msg_task = self.create_task(self._process_stt_messages())
 
     async def stop(self, frame: EndFrame):
         """Called when the session ends."""
+        if self._stt_msg_task and not self._stt_msg_task.done():
+            self._stt_msg_task.cancel()
         await super().stop(frame)
         await self._disconnect()
 
     async def cancel(self, frame: CancelFrame):
         """Called when the session is cancelled."""
+        if self._stt_msg_task and not self._stt_msg_task.done():
+            self._stt_msg_task.cancel()
         await super().cancel(frame)
         await self._disconnect()
 
-    async def run_stt(self, audio: bytes) -> AsyncGenerator[Frame, None]:
-        """Adds audio to the audio buffer and yields None."""
+    async def _connect(self) -> None:
+        """Connect to the STT service.
+
+        - Create STT client
+        - Register handlers for messages
+        - Connect to the client
+        """
+        # Log the event
+        logger.debug(f"{self} connecting to Speechmatics STT service")
+
+        # Update the audio sample rate
+        self._config.sample_rate = self._sample_rate
+
+        # STT client
+        self._client: VoiceAgentClient = VoiceAgentClient(
+            api_key=self._api_key,
+            url=self._base_url,
+            app=f"pipecat/{pipecat_version}",
+            config=self._config,
+        )
+
+        # Add message queue
+        def add_message(message: dict[str, Any]):
+            self._stt_msg_queue.put_nowait(message)
+
+        # Add listeners
+        self._client.on(AgentServerMessageType.ADD_PARTIAL_SEGMENT, add_message)
+        self._client.on(AgentServerMessageType.ADD_SEGMENT, add_message)
+
+        # Add listeners for VAD
+        if self._enable_vad:
+            self._client.on(AgentServerMessageType.START_OF_TURN, add_message)
+            self._client.on(AgentServerMessageType.END_OF_TURN, add_message)
+
+        # Speaker result listener
+        if self._config.enable_diarization:
+            self._client.on(AgentServerMessageType.SPEAKERS_RESULT, add_message)
+
+        # Other messages for debugging
+        self._client.on(AgentServerMessageType.ERROR, add_message)
+        self._client.on(AgentServerMessageType.WARNING, add_message)
+        self._client.on(AgentServerMessageType.INFO, add_message)
+        self._client.on(AgentServerMessageType.END_OF_TURN_PREDICTION, add_message)
+        self._client.on(AgentServerMessageType.END_OF_UTTERANCE, add_message)
+
+        # Connect to the client
+        try:
+            await self._client.connect()
+            logger.debug(f"{self} connected")
+        except Exception as e:
+            err_msg = f"{self} error connecting: {e}"
+            logger.error(err_msg)
+            self._client = None
+            await self.push_error(ErrorFrame(error=err_msg, fatal=True))
```

Comment:
Importantly, errors from service classes are never fatal. The reasoning being that developers might want to fail over to a backup service.

Also, the format is:
```suggestion
            await self.push_error(error_msg=f"Unable to connect to Speechmatics: {e}", exception=e)
```